test/celoBlockSignatureVerifier.js

```diff
@@ -0,0 +1,124 @@
+const { assert } = require("chai");
+const { tonelli } = require("./tonelli")
+const TestHelperContract = artifacts.require('TestHelper');
+const { blockHeaderRLP, blockHashPrefix, blockHashSuffix, blockHashBLSHints,
+    blockHashSignature, aggregatePublicKey, transactionMerkleKey, transactionMerkleNodes,
+    preimagePart } = require("./proofData");
+
+// START: taken from https://github.com/mrsmkl/b12-sol/tree/875d41c5c02a46ec9b2144dc9e8f48909f8c81aa/test
+let base = 0x1ae3a4617c510eac63b05c06ca1493b1a22d9f300f5138f1ef3622fba094800170b5d44300000008508c00000000001n
+let y1 = 0x001cefdc52b4e1eba6d3b6633bf15a765ca326aa36b6c0b5b1db375b6a5124fa540d200dfb56a6e58785e1aaaa63715bn
+let y2 = 0x01914a69c5102eff1f674f5d30afeec4bd7fb348ca3e52d96d182ad44fb82305c2fe3d3634a9591afd82de55559c8ea6n
+let x = 0x008848defe740a67c8fc6225bf87ff5485951e2caa9d41bb188282c8bd37cb5cd5481512ffcd394eeab9b16eb21be9efn
+
+function max(a,b) {
+  if (a < b) return b
+  else return a
+}
+
+function min(a,b) {
+  if (a > b) return b
+  else return a
+}
+
+function findY(x, greatest) {
+  let [a, b] = tonelli((x ** 3n + 1n) % base, base)
+  return [max(a,b), min(a,b)]
+}
+
+function uncompressSig(comp) {
+  let sig = [...comp].reverse()
+  let greatest = (sig[0] & 0x80) != 0
+  sig[0] = sig[0] & 0x7f
+  let x = BigInt("0x"+Buffer.from(sig).toString("hex"))
+  let [a, b] = tonelli((x ** 3n + 1n) % base, base)
+  let y = greatest ? max(a,b) : min(a,b)
+  // console.log(x, a, b, greatest ? max(a,b) : min(a,b), a < b, greatest)
+  return `0x${x.toString(16).padStart(128,0)}${y.toString(16).padStart(128,0)}`
+}
+
+async function makeHint(instance, blockHash) {
+  // let inner_hash = inner
+  // let extra_data = extra // counter, max nonsigners, epoch
+  let res = await instance.testHash(blockHash)
+  // console.log(res);
+  let arr = [...Buffer.from(res.substr(2), "hex")]
+  // console.log(arr.slice(0, 48))
+  let needed = arr.slice(0, 48).reverse()
+  // Parse to point
+  needed[0] = needed[0] & 0x01
+  let x = BigInt("0x" + Buffer.from(needed).toString("hex"))
+  let [y1, y2] = findY(x)
+  // console.log("x y1 y2", x.toString(16), y1.toString(16), y2.toString(16))
+  // let parsed_x = await instance.testParseToRandom(extra_data, inner_hash)
+  // console.log('parsed_x', parsed_x)
+  let hints = `0x${y1.toString(16).padStart(128, 0)}${y2.toString(16).padStart(128, 0)}`
+  // let point = await instance.testParseToG1Scaled(extra_data, inner_hash, hints)
+  // console.log('point', point)
+  return hints
+}
+// END: taken from https://github.com/mrsmkl/b12-sol/tree/875d41c5c02a46ec9b2144dc9e8f48909f8c81aa/test
+
+// Does not work in ganache yet. Works with normal Celo node.
+contract.skip("CeloBlockSignatureVerifier", function () {
```

Comment:
Consider a possibility of adding a condition to run these tests on celo and to skip them on all other chains